public/apps/configuration/utils/test/password-strength-bar.test.tsx

```diff
@@ -0,0 +1,94 @@
+/*
+ *   Copyright OpenSearch Contributors
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import { render, shallow } from 'enzyme';
+import React from 'react';
+import { PasswordStrengthBar } from '../password-strength-bar';
+
+describe('Password strength bar tests', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders', () => {
+    const component = shallow(<PasswordStrengthBar password="test" />);
+    expect(component).toMatchSnapshot();
+  });
+
+  it('verifies feedback, warning for very weak password', () => {
+    const component = render(<PasswordStrengthBar password="test" />);
+
+    const warning = component.find('[data-test-subj="password-strength-feedback-warning"]');
+    expect(warning.text()).toBe('This is a top-100 common password');
```

Comment:
IMO: `This is a top-100 common password` seems like an extra information. I would like to keep it minimal and relevant.

Reply:
okay, I have no more questions. Looks good to me.

Reply:
I was more wondering, if there is a case that the password is not an valid password (e.g. too short / missing a special character ... ), then what should we show here? Or all these msg should only be popped up when the password format meets the minimum requirements?

Reply:
This is an example I like from Wix: https://miro.medium.com/v2/resize:fit:1400/format:webp/1*v3N8tIHyKPa3Pz7lSnpPiA.gif

See the full article here: https://wix-ux.com/how-we-increased-our-users-passwords-strength-by-26-ab25efffa622

I've never seen a password set/reset UX where it tells me the password is too common. I've only seen UX that tells me the pw is too weak. Sometimes a site may also provide the minimum checklist if the website has explicit restrictions on character length, numbers and special characters. 

The warnings I think would be appropriate is if the user has CAPS LOCK on inadvertently.

Reply:
I see. Yea from what I've seen in their source code, I think suggestions seem more applicable to helping the user generate a stronger password. The only useful warning I've seen is "Common names and surnames are easy to guess", but otherwise suggestions should help more - @DarshitChanpura @cwperks I'll just remove the warnings block and keep the suggestions?

Reply:
@RyanL1997 regarding your question [here](https://github.com/opensearch-project/security-dashboards-plugin/pull/1762#discussion_r1560073276), that is actually a separate issue - see: https://github.com/opensearch-project/security-dashboards-plugin/issues/1234. Since the backend has two requirements, one which is regex matching, and one which is password strength via `zxcvbn` library, we need to add UX to satisfy both. This is just for `zxcvbn` library, and I will hopefully create another PR to add UX surrounding the regex requirements. However, I think it would be fair to say that _most_ of the time, if they satisfy strong password via `zxcvbn` they will also satisfy regex.

Reply:
@kamingleung can you help chime in here for final path forward? Thanks!

Reply:
As I mentioned in https://github.com/opensearch-project/security-dashboards-plugin/issues/1523#issuecomment-1656376784 and on Slack channel, a password strength checker is helpful to users when it can provide recommendations to make their passwords stronger. Only telling users their passwords WEAK or STRONG is rather vague and not actionable for users. 'This is a top-100 common password' is also in the same line of vagueness.

Instead of 'This is a top-100 common password', I would suggest we display recommendations such as:
"We recommend using passwords with random combinations of upper and lowercase letters, numbers, and special characters."
This way, it guides the users on what to do to make their password stronger.

Reply:
@derek-ho, based on @kamingleung's comments. I think this block: https://github.com/opensearch-project/security-dashboards-plugin/pull/1762/files#diff-ac97f3618a6ad790d8d2692a16dcd87421f4aee2de3f856167cd740505d31b52R60 might not be necessary. Instead, can we maintain a separate list of feedback messages so we don't rely on the ones coming from ZXCVBN for a more uniform user experience?